### PR TITLE
Defines Provider

### PR DESCRIPTION
**Pull request by dmrogers7**
Defines the `Provider` object structure for creating a kubernetes provider

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -269,6 +269,17 @@ paths:
...
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Object that describes the kubernetes provider"
+        required: true
+        schema:
+          $ref: "#/definitions/Provider"
```
Comment (billiford) [inline api/swagger.yaml:282]: Rename this to `KubernetesProvider` to allow for other providers in the future.

File: api/swagger.yaml [hunk 2]
```diff
@@ -295,4 +306,27 @@ paths:
           description: "Not Found"
         "500":
           description: "Internal Server Error"
-
+definitions:
```
Comment (billiford) [inline api/swagger.yaml:309]: The definition for `KubernetesProvider` should also contain the `permissions` key.

In JSON it looks like this:
```json
  "permissions": {
    "read": [
      "test-read-group"
    ],
    "write": [
      "test-write-group"
    ]
  }
  ```
In the code these are actually separate structs, so it's fine if you want to create them as separate definitions or define them as part of the same definition.

**Review by billiford — CHANGES_REQUESTED** [review]

File: api/swagger.yaml (hunk 1 above)
Comment (dmrogers7) [reply api/swagger.yaml]: done

File: api/swagger.yaml (hunk 2 above)
Comment (dmrogers7) [reply api/swagger.yaml:309]: Defined as a separate object for future reuse

**Review by billiford — APPROVED** [review]

**Pull request merged**